### PR TITLE
[wasm] Fix analyzer support in templates

### PR DESCRIPTION
**Pull request by radical**
- Add
  `[assembly:System.Runtime.Versioning.SupportedOSPlatform("browser")]`
  to the browser, and console templates. This would allow the analyzers,
  if enabled, to treat the assembly as one that will run only on
  browser.

- Populate `@(SupportedPlatform)` with only `browser`, for *wasm*
  projects, similar to https://github.com/dotnet/sdk/blob/fef8cedfb6b4ac85a7e135f3e4f155e29cdcbdf1/src/BlazorWasmSdk/Targets/Microsoft.NET.Sdk.BlazorWebAssembly.5_0.targets#L75-L79

- This also fixes the firefox CI build which regressed recently

### CODE REVIEW

File: src/mono/wasm/templates/templates/browser/Properties/AssemblyInfo.cs
Comment (maraf) [inline src/mono/wasm/templates/templates/browser/Properties/AssemblyInfo.cs:4]: Is this the best approach we can offer?

Comment (radical) [reply src/mono/wasm/templates/templates/browser/Properties/AssemblyInfo.cs:4]: IIUC, this is needed. I'm not sure how else we can do this, but open to suggestions!

**Comment by pavelsavara** [comment]
If this is the right way how to do it, then the samples\wasm should get the same treatment and then we could `<RunAnalyzers>true</RunAnalyzers>` in it.

**Comment by radical** [comment]
@akoeplinger does my implementation look correct?

I added it for the samples, but didn't enable the analyzers. We could try that.

**Comment by radical** [comment]
This is good to go, once we have an approval.

**Review by pavelsavara — APPROVED** [review]

**Review by steveisok — APPROVED** [review]

**Pull request merged**
